.github/workflows/images-legacy.yaml

```diff
@@ -168,11 +168,85 @@ jobs:
           path: image-digest
           retention-days: 1
 
+  # we need to build cilium-test separately because we run apt while building the image which requires qemu setup in order to avoid x86/arm64 binaries mixups
+  build-and-push-cilium-test:
+    if: ${{ github.repository == 'cilium/cilium' }}
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        include:
+          - name: cilium-test
+            dockerfile: ./images/cilium-test/Dockerfile
+
+    steps:
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
```

Comment:
Please pin a specific commit SHA, similarly to what is done in the other actions.